### ISSUE TITLE
consider adding on_return and on_error transform functions to timer_decorator

### ISSUE THREAD

**Issue by willkg** [willkg/markus#120]
I often want to decorate functions with `timer_decorator`, but want to provide different tags depending on the returned result or error.

What if we added `on_return` and `on_error` transform functions with defaults:

```
def on_return(tags, value, ret):
    return tags, value, ret
```

```
def on_error(tags, value, exc):
    return tags, value, exc
```

These would get called here after the function has completed:

https://github.com/willkg/markus/blob/a3a5db6b9635fa04ceb66f4b2c11058b4e09ba17/src/markus/main.py#L515

**Comment by willkg** [comment]
@jwhitlock Have you ever needed something like this?

**Comment by jwhitlock** [comment]
No, I haven't needed something like that, and I usually don't reach for `timer_decorator`

I've been using [codetiming](https://github.com/realpython/codetiming) recently like this:

```python
with Timer(logger=None) as something_timer:
    ret = do_something()
log_data = {
   something_s = round(something_timer.last, 3)
}
log.info("Ran something", extra=log_data)
markus.timing('ran.something', something_timer.last)
```

I can't really recommend my method or `codetiming` (main benefit - someone else maintains it). But usually, when I time something, I want to log it, not just send metrics.
